src/helpers/privacy_budget.rs

```diff
@@ -0,0 +1,80 @@
+/// This module implements privacy budget using a redis cache
+/// It expects a vector of encrypted blobs alongwith budget to be consumed.
+/// After increasing the budget consumed by all the blobs, it returns blobs which are below ``TOTAL_PRIVACY_BUDGET``
+
+use redis::Commands;
+use std::env;
+
+const TOTAL_PRIVACY_BUDGET: i32 = 100;
+
+#[must_use]
+pub fn get_eligible_packets_with_budget(packets: &[String], budget: i32) -> Vec<String> {
+    increase_budget_consumed(packets, budget);
+    get_eligible_packets(packets)
+}
+
+fn increase_budget_consumed(packets: &[String], budget: i32) {
+    let mut conn = connect();
+    for packet in packets.iter() {
+        let _: () = conn
+            .incr(packet.to_string(), budget)
+            .expect("failed to execute INCR for a packet");
+    }
+}
+fn get_eligible_packets(packets: &[String]) -> Vec<String> {
+    let mut conn = connect();
+    let mut eligible_packets: Vec<String> = Vec::new();
+    for packet in packets.iter() {
+        let budget_consumed: i32 = redis::cmd("GET")
+            .arg(packet)
+            .query(&mut conn)
+            .expect("Failed to GET budget for packet");
+
+        if budget_consumed < TOTAL_PRIVACY_BUDGET {
+            eligible_packets.push(packet.to_string());
+        }
+    }
+    eligible_packets
+}
+
+fn connect() -> redis::Connection {
```

Comment:
This should produce an object (a new struct) that hosts the above functions.  Then the connection can be held open by the caller until they are done.